### ISSUE TITLE
Update `matchHost` to support ElasticSearch hosts

### ISSUE THREAD

**Issue by matthew-andrews** [mhart/aws4#8]
This regular expression will work for ElasticSearch…  Would you be open to this as a PR?

```js
/^(?:[^\.]+\.)?([^\.]+)\.?([^\.]*)\.amazonaws\.com$/
```

Example ElasticSearch hostname:-
search-next-search-aaaa00aaaa0aaa0aaaaaaa0aaa.us-east-1.es.amazonaws.com

On this line:-
https://github.com/mhart/aws4/blob/master/aws4.js#L46

Diff:-
https://github.com/mhart/aws4/compare/master...matthew-andrews:patch-1

**Comment by matthew-andrews** [comment]
Hmm not whilst that regular expression works, it's the wrong way round

**Comment by mhart** [comment]
You realise that you can set the `host` and the `service` separately right? Like, you shouldn't have to rely on the host parsing.

At least, I'm pretty sure that should work with the current code.

**Comment by mhart** [comment]
So do:
```js
aws4.sign({
  service: 'es',
  region: 'us-east-1',
  host: 'search-next-search-aaaa00aaaa0aaa0aaaaaaa0aaa.us-east-1.es.amazonaws.com',
})
```

**Comment by mhart** [comment]
(Obviously having the host parse correctly is a nice-to-have and I can look at it when I get the time – but just saying this should work out of the box now)

**Comment by matthew-andrews** [comment]
yup i got it working with that but we use Elasticsearch in a couple of different regions so was easier to ‘fix’ your library to auto-detect the region

**Comment by mhart** [comment]
Ah ok, so you just wanted to switch based on the host string? I mean, I would've thought this was easier than patching the library and maintaining a branch :smile_cat: 
```js
aws4.sign({
  service: 'es',
  region: host.split('.')[1],
  host: host,
})
```
But anyway, will look at this when I get the time.

**Comment by matthew-andrews** [comment]
I was hoping to contribute back :-)

I've got a unit test for this case, I'll make a new PR in a few moments.

**Comment by mhart** [comment]
Awesome, thanks! When you do – if there's some es-specific logic in there, just add a comment like this if you can: https://github.com/mhart/aws4/blob/master/aws4.js#L33

**Comment by mhart** [comment]
Closed by #10
